### PR TITLE
[ONNX] Add linspace symbolic

### PR DESCRIPTION
**Pull request by shubhambhokare1**
* Adds support for linspace op 
* Modifies arange symbolic in opset 9 to replicate the same behavior in which dtype is determined (similar to opset 11) as in https://pytorch.org/docs/stable/generated/torch.arange.html
* Enabled some arange unit tests which were disabled for opset 9

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -1935,17 +1935,79 @@ def forward(self, start, end):
...
+            def forward(self, start, end):
+                out_t = torch.tensor([1], dtype=torch.float)
+                return torch.arange(start.size(0), end, 1.5, out=out_t)
+
+        x = torch.randn(2, 3, 4)
+        y = torch.tensor(8.5, dtype=torch.float)
+        self.run_test(ArangeModelStep(), (x, y))
+
+    @skipIfUnsupportedMinOpsetVersion(9)
+    def test_arange_with_floats(self):
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1969]: could you validate negative steps?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2692,53 +2692,57 @@ def arange(g, *args):
...
         return dtype
 
-    if len(args) == 2:
-        # aten::arange(Scalar end, Tensor out)
-        end = sym_help._unsqueeze_helper(g, args[0], [0])
-        dtype = 4  # default to int64
-        arange_tensor = sym_help._squeeze_helper(g, nonzero(g, ones(g, end, dtype, None, None)), [1])
+    def _float_step_convert(range_tensor):
+        if sym_help._is_fp(range_tensor):
+            range_tensor = g.op("Cast", g.op("Ceil", range_tensor), to_i=sym_help.scalar_type_to_onnx[4])
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2699]: why is the cast to 4 required?

**Review by BowenBao — COMMENTED** [review]
looks good overall, please rebase

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2701]: This is for cases when steps are floats, their ceils are also floats -> hence range tensor is float as well.
Now this range tensor is called by ones() and subsequently as ConstantOfShape()'s sizes paramater, which raises an error if this value is not an int64, hence the cast node.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (shubhambhokare1) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:1968]: Add tests for negative steps

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -530,6 +530,12 @@ def _get_arange_dtype(dtype):
         raise NotImplementedError("Unknown aten::arange signature taking " + str(len(args)) + " arguments.")
     return arange_tensor
 
+def linspace(g, start, end, steps, dtype, layout, device, pin_memory):
+    dtype = sym_help._maybe_get_const(dtype, "i")
+    step = div(g, sub(g, end, start), sub(g, steps, g.op("Constant", value_t=torch.tensor(1, dtype=torch.int64))))
+    end_epsilon = g.op("Add", step, end)
+    type, end, start, step = sym_help._arange_cast_helper(g, start=start, end=end_epsilon, step=step, dtype=dtype)
+    return g.op("Range", start, end, step)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py:538]: There is some duplication of code with opset 9 linspace. Seems a `arange` helper may come in handy such that only need to define linspace in opset 9?

Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py]: I think we should be fine just using the opset 9 symbolic. Nice catch :)

Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py]: Updated by adding helper function

**Review by BowenBao — APPROVED** [review]

**Pull request merged**